pkg/executor/core/executor.go

```diff
@@ -173,13 +211,65 @@ func (executor Executor) verifyReference(ctx context.Context, subjectRef common.
 	return types.VerifyResult{IsSuccess: isSuccess, VerifierReports: verifyResults}
 }
 
+// verifyReferenceForRegoPolicy verifies the referenced artifact with results
+// used for Rego-based policy enforcer.
+func (executor Executor) verifyReferenceForRegoPolicy(ctx context.Context, subjectRef common.Reference, referenceDesc ocispecs.ReferenceDescriptor, referrerStore referrerstore.ReferrerStore) (types.NestedVerifierReport, error) {
+	nestedReport := types.NestedVerifierReport{
+		Subject:         subjectRef.String(),
+		ArtifactType:    referenceDesc.ArtifactType,
+		ReferenceDigest: referenceDesc.Digest.String(),
+		VerifierReports: make([]vt.VerifierResult, 0),
+		NestedReports:   make([]types.NestedVerifierReport, 0),
+	}
+	var mu sync.Mutex
+	eg, errCtx := errgroup.WithContext(ctx)
+
+	eg.Go(func() error {
+		return executor.addNestedReports(errCtx, referenceDesc, subjectRef, &nestedReport)
+	})
+
+	for _, verifier := range executor.Verifiers {
+		if !verifier.CanVerify(ctx, referenceDesc) {
+			continue
+		}
+		verifier := verifier
+		eg.Go(func() error {
+			var verifierReport vt.VerifierResult
+			verifierStartTime := time.Now()
+			verifierResult, err := verifier.Verify(errCtx, subjectRef, referenceDesc, referrerStore)
+			if err != nil {
+				verifierReport = vt.VerifierResult{
+					IsSuccess: false,
+					Name:      verifier.Name(),
+					Message:   fmt.Sprintf("an error thrown by the verifier: %v", err)}
+			} else {
+				verifierReport = vt.NewVerifierResult(verifierResult)
+			}
+
+			mu.Lock()
+			nestedReport.VerifierReports = append(nestedReport.VerifierReports, verifierReport)
+			mu.Unlock()
+
+			metrics.ReportVerifierDuration(errCtx, time.Since(verifierStartTime).Milliseconds(), verifier.Name(), subjectRef.String(), verifierReport.IsSuccess, err != nil)
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return types.NestedVerifierReport{}, err
+	}
+	return nestedReport, nil
+}
+
+// addNestedVerifierResult adds the nested verifier result to the parent verify
+// result used for Json-based policy enforcer.
 func (executor Executor) addNestedVerifierResult(ctx context.Context, referenceDesc ocispecs.ReferenceDescriptor, subjectRef common.Reference, verifyResult *vr.VerifierResult) {
 	verifyParameters := e.VerifyParameters{
 		Subject:        fmt.Sprintf("%s@%s", subjectRef.Path, referenceDesc.Digest),
 		ReferenceTypes: []string{"*"},
 	}
 
-	nestedVerifyResult, err := executor.VerifySubject(ctx, verifyParameters)
+	nestedVerifyResult, err := executor.verifySubjectWithJSONPolicy(ctx, verifyParameters)
```

Comment:
should we rename with with "josnPolicy" in name? is there a rego equivalent method?